parlai/core/torch_generator_agent.py

```diff
@@ -635,12 +635,10 @@ def train_step(self, batch):
         """
         Train on a single batch of examples.
         """
-        if batch.text_vec is not None:
-            batchsize = batch.text_vec.size(0)
-        elif batch.image is not None:
-            batchsize = len(batch.image)
         # helps with memory usage
-        self._init_cuda_buffer(batchsize, self.truncate or 256)
+        # note we want to use the opt's batchsize instead of the observed batch size
+        # in case dynamic batching is in use
+        self._init_cuda_buffer(self.opt['batchsize'], self.label_truncate or 256)
```

Comment:
why did this change to label_truncate?

Reply:
It should have always been label truncate.

Reply:
Actually, I'm wrong. Hm.